services/sidecar/src/simcore_service_sidecar/executor.py

```diff
@@ -164,6 +166,12 @@ async def _process_task_input(self, port: node_ports.Port, input_ports: Dict):
                     final_path,
                     path,
                 )
+                # check if the file is a zip, in that case extract all
+                if zipfile.is_zipfile(final_path):
+                    with zipfile.ZipFile(final_path, "r") as zip_obj:
+                        zip_obj.extractall(final_path.parents[0])
```

Comment:
This is not the python library that is tremendously slow, right?

Reply:
I think it might just be it. Alternatives?

Reply:
check thisone https://www.peterbe.com/plog/fastest-way-to-unzip-a-zip-file-in-python

Reply:
no issues then with the zip?